src/core/control/tools/TextEditorContextMenu.cpp

```diff
@@ -0,0 +1,629 @@
+};
+void tglSecToolbar(GtkButton* src, TextEditorContextMenu* tecm) { tecm->toggleSecondaryToolbar(); }
+
+void tglBoldStyle(GtkButton* src, TextEditorContextMenu* tecm) { tecm->toggleWeight(PANGO_WEIGHT_BOLD); }
+void tglItalicStyle(GtkButton* src, TextEditorContextMenu* tecm) { tecm->toggleStyle(PANGO_STYLE_ITALIC); }
+void tglUnderlineStyle(GtkButton* src, TextEditorContextMenu* tecm) { tecm->toggleUnderline(PANGO_UNDERLINE_SINGLE); }
+
+
+void toggleWeightThinClb(GtkToggleButton* src, TextEditorContextMenu* tecm) { tecm->toggleWeight(PANGO_WEIGHT_THIN); }
+void toggleWeightBookClb(GtkToggleButton* src, TextEditorContextMenu* tecm) { tecm->toggleWeight(PANGO_WEIGHT_BOOK); }
+void toggleWeightBoldClb(GtkToggleButton* src, TextEditorContextMenu* tecm) { tecm->toggleWeight(PANGO_WEIGHT_BOLD); }
+
+void toggleStyleItalicClb(GtkToggleButton* src, TextEditorContextMenu* tecm) { tecm->toggleStyle(PANGO_STYLE_ITALIC); }
+void toggleStyleObliqueClb(GtkToggleButton* src, TextEditorContextMenu* tecm) {
+    tecm->toggleStyle(PANGO_STYLE_OBLIQUE);
+}
+void toggleUnderlineSingleClb(GtkToggleButton* src, TextEditorContextMenu* tecm) {
+    tecm->toggleUnderline(PANGO_UNDERLINE_SINGLE);
+}
+void toggleUnderlineSquiggleClb(GtkToggleButton* src, TextEditorContextMenu* tecm) {
+    tecm->toggleUnderline(PANGO_UNDERLINE_ERROR);
+}
+void toggleUnderlineDoubleClb(GtkToggleButton* src, TextEditorContextMenu* tecm) {
+    tecm->toggleUnderline(PANGO_UNDERLINE_DOUBLE);
+}
+void toggleStrikethroughClb(GtkToggleButton* src, TextEditorContextMenu* tecm) { tecm->toggleStrikethrough(TRUE); }
+void toggleOverlineSingleClb(GtkToggleButton* src, TextEditorContextMenu* tecm) {
+    tecm->toggleOverline(PANGO_OVERLINE_SINGLE);
+}
+
+void toggleSuperScriptClb(GtkToggleButton* src, TextEditorContextMenu* tecm) {
+    tecm->toggleScriptRise(5 * PANGO_SCALE);
+}
+void toggleSubScriptClb(GtkToggleButton* src, TextEditorContextMenu* tecm) {
+    tecm->toggleScriptRise(-(5 * PANGO_SCALE));
+}
+void toggleRemoveStyles(GtkButton* btn, TextEditorContextMenu* tecm) { tecm->removeAllAttributes(); }
+
+TextEditorContextMenu::TextEditorContextMenu(Control* control, TextEditor* editor, XojPageView* pageView,
+                                             GtkWidget* xournalWidget):
+        control(control), editor(editor), pageView(pageView), xournalWidget(xournalWidget) {
+    // Only for debugging
+    std::cout << "TextEditorContextMenu created!" << std::endl;
+    this->create();
+}
+
+TextEditorContextMenu::~TextEditorContextMenu() {
+    gtk_widget_destroy(GTK_WIDGET(this->fontBtn));
+    // gtk_widget_destroy(GTK_WIDGET(this->ftColorIcon));
+    gtk_widget_destroy(GTK_WIDGET(this->ftColorBtn));
+    // gtk_widget_destroy(GTK_WIDGET(this->bgColorIcon));
+    gtk_widget_destroy(GTK_WIDGET(this->bgColorBtn));
+    gtk_widget_destroy(GTK_WIDGET(this->alignLeftTgl));
+    gtk_widget_destroy(GTK_WIDGET(this->alignCenterTgl));
+    gtk_widget_destroy(GTK_WIDGET(this->alignRightTgl));
```

Comment:
Hi, 
I wrapped the Gtk-Elements into a GObjectSPtr like this (I think thats what you mean with using RAII or?):
```c++
xoj::util::GObjectSPtr<GtkFontButton> fontBtn;
```
But now when the context menu is destroyed I get the following warning:
``` bash
(xournalpp:23915): GLib-GObject-CRITICAL **: 12:35:04.105: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
```
I think its due to destroying the object using `g_object_unref()` instead of `gtk_widget_destroy()`. Probably I'm using GObjectSPtr wronly. I would appreciate if you could give me an example on how to use RAII wrappers correctly.

Reply:
As mentioned here  https://github.com/xournalpp/xournalpp/pull/4938#discussion_r1541550574, i think RAII is not required for these Widgets, therefore I removed it again.

Instead, I added all the missing widgets to the deconstructor. (Should be unnecessary as the parent widget should destroy all child widgets)